pipelines/update_exposure_tercile.py

```diff
@@ -0,0 +1,120 @@
+import sys
+from datetime import datetime, timedelta
+
+import numpy as np
+import pandas as pd
+from sqlalchemy import text
+
+from src.utils import database
+
+ROLL_WINDOW = 7
+
+ROLLING_QUERY = text(
+    f"""
+    WITH target_dates AS (
+        SELECT
+            pcode,
+            adm_level,
+            valid_date,
+            sum as original_sum
+        FROM app.floodscan_exposure
+        WHERE EXTRACT(MONTH FROM valid_date) = :month
+            AND EXTRACT(DAY FROM valid_date) = :day
+    ),
+    date_ranges AS (
+        SELECT
+            t.pcode,
+            t.adm_level,
+            t.valid_date as target_date,
+            t.original_sum as sum,
+            d.valid_date,
+            d.sum as daily_sum
+        FROM target_dates t
+        JOIN app.floodscan_exposure d
+            ON d.pcode = t.pcode
+            AND d.valid_date BETWEEN t.valid_date - INTERVAL '{ROLL_WINDOW-1} days' AND t.valid_date
+    )
+    SELECT
+        pcode,
+        adm_level,
+        target_date as valid_date,
+        AVG(daily_sum) as rolling_avg
+    FROM date_ranges
+    GROUP BY pcode, adm_level, target_date
+    ORDER BY pcode, target_date;
+    """
+)
+
+
+def assign_tercile(row, boundaries):
+    """
+    Assign tercile values based on boundaries.
+    Returns:
+        -1: Below lower tercile
+        0: Between terciles
+        1: Above upper tercile
+    """
+    pcode_bounds = boundaries.loc[row["pcode"]]
+    value = row["rolling_avg"]
+    if value < pcode_bounds["lower_tercile"]:
+        return -1
+    elif value <= pcode_bounds["upper_tercile"]:
+        return 0
+    else:
+        return 1
+
+
+if __name__ == "__main__":
+    target_date = datetime.today() - timedelta(days=1)
+    engine = database.get_engine()
+
+    print(f"Computing terciles as of {target_date.strftime('%Y-%m-%d')}")
+    print(f"Using {ROLL_WINDOW}-day rolling average")
+
+    # Get data and calculate rolling averages
+    try:
+        with engine.connect() as con:
+            print("Calculating rolling averages and getting data from db...")
+            df_all = pd.read_sql_query(
+                ROLLING_QUERY,
+                con,
+                params={"month": target_date.month, "day": target_date.day},
+            )
+    except Exception as e:
+        print(f"Error querying database: {e}")
+        sys.exit(1)
+
+    if df_all.empty:
+        print(
+            f"No data retrieved from database for {target_date.strftime('%Y-%m-%d')}"  # noqa
+        )
+        sys.exit(0)
+
+    print("Computing terciles...")
+    tercile_boundaries = df_all.groupby("pcode")["rolling_avg"].agg(
+        lower_tercile=lambda x: np.percentile(x, 33.33),
```

Comment:
You think worth setting as `100 / 3`? Guess it's a pretty small discrepancy so doesn't really matter though, but would make more configurable in the future to different quantiles.

Reply:
To be fully configurable, it makes the following quantiles a bit more challenging though, no? ie. we need some kind of loop where we have `100 / n_quantiles * i`? I just worry it'd be a bit less readable for functionality that we don't need yet.

Reply:
Yeah we can just leave it for now. I think if we wanted it more robust to different quantiles we can just sort of to the calculation backwards, i.e. calculate the percentile of the value that year (e.g. 30th), and see what bin that falls in (lower tercile, second bottom quintile, etc).